src/main/kotlin/org/rust/ide/console/RsConsoleView.kt

```diff
@@ -73,9 +78,54 @@ class RsConsoleView(project: Project) : LanguageConsoleImpl(project, VIRTUAL_FIL
         if (codeFragment != null) {
             codeFragmentContext.addToContext(lastCommandContext)
             codeFragmentContext.updateContext(project, codeFragment)
+            variablesView?.rebuild()
         }
     }
 
+    private fun splitWindow() {
```

Comment:
For now, the logic is not obvious IMO. Maybe get rid of `showVariables` and do everything in-place?  
```Kotlin
private fun showVariables() {
    val console = getComponent(0)
    if (variablesView == null) {
        variablesView = RsConsoleVariablesView(project, codeFragmentContext)
        removeAll()
        // ...
    }
}
```